src/query/storages/common/cache/src/cache_items.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use bytes::Bytes;
+pub use databend_common_catalog::plan::PartStatistics;
+pub use databend_common_catalog::plan::Partitions;
+pub use databend_common_catalog::table::Table;
+use databend_common_exception::ErrorCode;
+pub use databend_storages_common_index::filters::Xor8Filter;
+pub use databend_storages_common_index::BloomIndexMeta;
+pub use databend_storages_common_index::InvertedIndexFile;
+pub use databend_storages_common_index::InvertedIndexMeta;
+pub use databend_storages_common_table_meta::meta::column_oriented_segment::ColumnOrientedSegment;
+pub use databend_storages_common_table_meta::meta::BlockMeta;
+pub use databend_storages_common_table_meta::meta::CompactSegmentInfo;
+pub use databend_storages_common_table_meta::meta::SegmentInfo;
+pub use databend_storages_common_table_meta::meta::TableSnapshot;
+pub use databend_storages_common_table_meta::meta::TableSnapshotStatistics;
+pub use parquet::file::metadata::ParquetMetaData;
+
+use crate::HybridCache;
+
+pub struct ColumnData(Bytes);
+
+impl ColumnData {
+    pub fn from_merge_io_read_result(bytes: Bytes) -> Self {
+        // Bytes are from merge read result, may refer to a large chunk of memory;
+        // Caching this large buffer wastes memory, so we need to copy it out.
```

Comment:
I don't get the comments.
If `bytes` are continuous, maybe we don't need to copy it out?

Reply:
I get it, so bytes are sliced.